recipes/tsnecuda/meta.yaml

```diff
@@ -0,0 +1,76 @@
+{% set version = "3.0.0" %}
+
+package:
+  name: tsnecuda
+  version: {{ version }}
+
+source:
+  - url: https://github.com/CannyLab/tsne-cuda/archive/refs/tags/{{ version }}.tar.gz
+    sha256: 6f5a0c5c0c54a4a74837e0e84fa37396ca6912a2031bdf6e846d5c01254e3e2c
+    patches:
+      # do not try to detect (or set) "build number" from git;
+      # we're only building released versions anyway
+      - patches/0001-do-not-try-to-detect-set-build-number.patch
+      - patches/0002-add-patch_level-to-version_string.patch
+      - patches/0003-adapt-path-that-depends-on-invocation-in-a-build-dir.patch
+      - patches/0004-fix-version-insertion-in-src-exe-main.cu.patch
+  # include submodule (not in github tarball due to dear-github/dear-github#214), c.f.
+  # https://github.com/CannyLab/tsne-cuda/tree/{{ version }}/third_party
+  - git_url: https://github.com/jarro2783/cxxopts.git
+    git_rev: c713b44d92e4ea2569101fa6bdf60afa311b29a3
+    folder: third_party/cxxopts
+
+
+build:
+  number: 0
+  # only linux for the moment
+  skip: true  # [not linux]
+  skip: true  # [cuda_compiler_version == "None"]
+  string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - cmake
+    - make
+  host:
+    - python
+    - pip
+    # package uses zmq hpp-headers
+    - cppzmq
+    - gflags
+    - gmock
+    - libblas
+    - liblapack
```

Comment:
I didn't verify that those are absolutely necessary, but the upstream CMake script looks for them.

Reply:
So, either you need Intel MKL _or_ you need BLAS/LAPACK (see https://github.com/CannyLab/tsne-cuda/blob/b740a7d46a07ca9415f072001839fb66a582a3fa/CMakeLists.txt#L232). It's actually faster to use MKL on most intel-based machines, and even on some AMD machines than a poorly tuned ATLAS install, or a base OPENBLAS install. Ideally, we allow conda users to select for this on their own, but I have no clue how to make that happen (or if the MKL packages in conda provide libblas/liblapack making this irrelevant).

Reply:
>  but I have no clue how to make that happen (or if the MKL packages in conda provide libblas/liblapack making this irrelevant).

Conda has a fairly sophisticated mechanism for this. Libraries are compiled against netlib blas/lapack, which can be replaced at runtime with any of openblas, MKL or blis. Happy to explain in more detail, if you want.